packages/core/test/unit/checker/checker-facade.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { CheckStatus } from '@stryker-mutator/api/check';
+import { factory } from '@stryker-mutator/test-helpers';
+
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { CheckerFacade } from '../../../src/checker/checker-facade.js';
+import { CheckerResource } from '../../../src/checker/index.js';
+
+import { createMutantRunPlan } from '../../helpers/producers.js';
+
+describe.only(CheckerFacade.name, () => {
+  let innerChecker: sinon.SinonStubbedInstance<CheckerResource>;
+  let sut: CheckerFacade;
+
+  beforeEach(() => {
+    innerChecker = {
+      init: sinon.stub(),
+      group: sinon.stub(),
+      check: sinon.stub(),
+      dispose: sinon.stub(),
+    };
+    const checkers = [innerChecker];
+    sut = new CheckerFacade(() => checkers.shift()!);
```

Comment:
Why the use of an array here?

Reply:
Because I thought the lambda paramater would need to return `null` after all checkers where loaded, but it seems that isn't the case. I would change this in next commit.

Reply:
No, indeed. The factory method is called whenever the decorator or facade wants to create a new inner resource. Thanks for updating!